src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1845,8 +1828,8 @@ export default class AircraftModel {
 
         if (shouldMoveToNextFix) {
             if (!this.fms.hasNextWaypoint()) {
-                // we've hit this block becuase and aircraft is about to fly over the last waypoint in its flightPlan
-                this.pilot.maintainPresentHeading(this.heading);
+                // we've hit this block because and aircraft is about to fly over the last waypoint in its flightPlan
+                this.pilot.maintainPresentHeading(headingToWaypoint);
```

Comment:
that's not the right value though, we definitely can't do this. Not sure what the test issue was, but this will need to be `this.heading`.

Reply:
Also, in the app, it seems like the headings are not being assigned properly, because if I ask an airplane who has passed their last fix what heading they are on, they say `assigned heading NaN`. Not sure how they ended up with that, but they do.

Reply:
Looks like `Pilot.maintainPresentHeading()` takes the aircraftModel, which it inspects to find the aircraft's current heading, rather than taking the heading. Sounds dumb (and kinda is), but the Pilot so frequently has to inspect the aircraft to execute an action that doing this has become a relatively common pattern with it. Hence why the Pilot needs to die (#1177).

Thus, this should be changed to `this.pilot.maintainPresentHeading(this);`. :scream: :grimacing:

Reply:
done